yaksh/views.py

```diff
@@ -326,6 +326,56 @@ def add_quiz(request, quiz_id=None, course_id=None):
     )
 
 
+@login_required
+@email_verified
+def add_exercise(request, quiz_id=None, course_id=None):
+    user = request.user
+    ci = RequestContext(request)
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this course !')
+    if quiz_id:
+        quiz = get_object_or_404(Quiz, pk=quiz_id)
+        if quiz.creator != user and not course_id:
+            raise Http404('This quiz does not belong to you')
+    else:
+        quiz = None
+    if course_id:
+        course = get_object_or_404(Course, pk=course_id)
+        if not course.is_creator(user) and not course.is_teacher(user):
+            raise Http404('This quiz does not belong to you')
+
+    context = {}
+    if request.method == "POST":
+        form = ExerciseForm(request.POST, instance=quiz)
+        if form.is_valid():
+            if quiz is None:
+                form.instance.creator = user
+            quiz = form.save(commit=False)
+            quiz.is_exercise = True
+            quiz.time_between_attempts = 0
+            quiz.weightage = 0
+            quiz.allow_skip = False
+            quiz.attempts_allowed = -1
+            quiz.duration = 1000
```

Comment:
Problem with this is no exercise is going to work after 1000 mins (or appx 17 hrs). This is a bad solution, since exercises should inherently be timeless.

Reply:
Also for God mode, the moderator should be able to go to the next question, irrespective of the question duration.

Reply:
Yes I agree not a good solution, but this is just to replace database default value.
But this will work timeless as in views there is no time check.
Also if the user starts the exercise and is inactive for 17 hours the exercise may end, but then we have infinite attempts allowed as it is an exercise.

God mode and user mode will work as our quiz works because while cloning is_exercise is False.
This needs to be True for User mode. User mode and God mode are currently disabled for Exercise.  Will enable it.